RFS/src/test/java/com/rfs/worker/SnapshotStepTest.java

```diff
@@ -0,0 +1,226 @@
+package com.rfs.worker;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.api.Test;
+
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.rfs.cms.CmsClient;
+import com.rfs.cms.CmsEntry.Snapshot;
+import com.rfs.cms.CmsEntry.SnapshotStatus;
+import com.rfs.common.SnapshotCreator;
+import com.rfs.common.SnapshotCreator.SnapshotCreationFailed;
+import com.rfs.worker.SnapshotStep.ExitPhaseSnapshotFailed;
+import com.rfs.worker.SnapshotStep.ExitPhaseSuccess;
+import com.rfs.worker.SnapshotStep.WaitForSnapshot;
+
+import static org.mockito.Mockito.*;
+
+
+@ExtendWith(MockitoExtension.class)
+public class SnapshotStepTest {
+
+    @Test
+    void EnterPhase_run_AsExpected() {
+        // Set up the test
+        GlobalState globalState = Mockito.mock(GlobalState.class);
+        CmsClient cmsClient = Mockito.mock(CmsClient.class);
+        SnapshotCreator snapshotCreator = Mockito.mock(SnapshotCreator.class);
+        Snapshot snapshotEntry = Mockito.mock(Snapshot.class);
```

Comment:
Move all of these into a shared setup, all tests seem to be reusing these same blank mocks